### PR TITLE
Fix Lumen, add PostProcessConfig.

### PR DESCRIPTION
**Pull request by MarcelPiNacy-CVC**
This PR fixes Lumen being disabled due to EnableExtendedFeatures() and adds a new class to set post process settings and show flags of SceneCaptureSensor.

- - -
This change is [Reviewable](https://reviewable.io/reviews/carla-simulator/carla/7705)

### CODE REVIEW

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/SceneCaptureSensor.cpp [hunk 1]
```diff
@@ -55,10 +55,10 @@ ASceneCaptureSensor::ASceneCaptureSensor(const FObjectInitializer &ObjectInitial
   CaptureRenderTarget->AddressX = TextureAddress::TA_Clamp;
   CaptureRenderTarget->AddressY = TextureAddress::TA_Clamp;
 
-  CaptureComponent2D = CreateDefaultSubobject<USceneCaptureComponent2D_CARLA>(
-      FName(*FString::Printf(TEXT("USceneCaptureComponent2D_CARLA_%d"), SCENE_CAPTURE_COUNTER)));
+  CaptureComponent2D = CreateDefaultSubobject<USceneCaptureComponent2D>(
+      FName(*FString::Printf(TEXT("USceneCaptureComponent2D%d"), SCENE_CAPTURE_COUNTER)));
   check(CaptureComponent2D != nullptr);
-  CaptureComponent2D->ViewActor = this;
+  //CaptureComponent2D->ViewActor = this;
```
Comment (Blyron) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/SceneCaptureSensor.cpp:61]: Remove the commentary

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/SceneCaptureSensor.cpp [hunk 2]
```diff
@@ -791,161 +805,4 @@ namespace SceneCaptureSensor_local_ns {
...
-    ShowFlags.SetVisualizeSSR(false);
-    ShowFlags.SetVisualizeSSS(false);
-    // ShowFlags.SetVolumeLightingSamples(false);
-    // ShowFlags.SetVolumes(false);
-    // ShowFlags.SetWidgetComponents(false);
-    // ShowFlags.SetWireframe(false);
-  }
-
-} // namespace SceneCaptureSensor_local_ns
+} // namespace SceneCaptureSensor_local_ns
```
Comment (Blyron) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/SceneCaptureSensor.cpp]: EOF

**Review by Blyron — CHANGES_REQUESTED** [review]
:lgtm:

> Reviewed 5 of 8 files at r1, 1 of 2 files at r2, 1 of 2 files at r3, 4 of 4 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/carla-simulator/carla/7705#-:-Nyyzt-57tOlK1_r9Ami:bnfp4nl)* status: all files reviewed, 5 unresolved discussions (waiting on @MarcelPiNacy-CVC)

___
*[`Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.h` line 46 at r6](https://reviewable.io/reviews/carla-simulator/carla/7705#-Nyz-49ZDn4oMgVSbQaE:-Nyz-49ZDn4oMgVSbQaF:b-qzggrs) ([raw file](https://github.com/carla-simulator/carla/blob/1d1d13972097a0a1e9c82797dc2bb8f78cd3570b/Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.h#L46)):*
> ```objc
> 
> 
> };
> ```

Add EOF
___
*[`Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.cpp` line 13 at r6](https://reviewable.io/reviews/carla-simulator/carla/7705#-Nyz-7nZ8kLcBDUSIqGZ:-Nyz-7nZ8kLcBDUSIqG_:bf7yljx) ([raw file](https://github.com/carla-simulator/carla/blob/1d1d13972097a0a1e9c82797dc2bb8f78cd3570b/Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.cpp#L13)):*
> ```cpp
> ASkyBase::ASkyBase(
> 	const FObjectInitializer& ObjectInitializer) :
> 	Super(ObjectInitializer),
> ```

The indentation must be spaces, not tabs
___
*[`Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.cpp` line 15 at r6](https://reviewable.io/reviews/carla-simulator/carla/7705#-Nyz-IagAGud6lfUcx6A:-Nyz-IagAGud6lfUcx6B:btw9k2l) ([raw file](https://github.com/carla-simulator/carla/blob/1d1d13972097a0a1e9c82797dc2bb8f78cd3570b/Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.cpp#L15)):*
> ```cpp
> 	Super(ObjectInitializer),
> 	PostProcessComponent(CreateDefaultSubobject<UPostProcessComponent>("ASkyBase::PostProcessComponent")),
> 	ExponentialHeightFogComponent(CreateDefaultSubobject<UExponentialHeightFogComponent>("ASkyBase::ExponentialHeightFogComponent")),
> ```

This is not standard coding. 
The standard coding for UE in component intialization must be like
AClass::AClass(const FObjectIni... ObjInit) :Super(ObjInit)
{
 ComponentVar = CreateDefaultSubbject<UClass>("ComponentVar")
}

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/SceneCaptureSensor.cpp [hunk 3]
```diff
@@ -713,6 +727,45 @@ namespace SceneCaptureSensor_local_ns {
...
+    PostProcessSettings.bOverride_LumenReflectionQuality = true;
+    PostProcessSettings.LumenReflectionQuality = 1.0f;
+    PostProcessSettings.bOverride_LumenRayLightingMode = true;
+    PostProcessSettings.LumenRayLightingMode = ELumenRayLightingModeOverride::Default;
+    PostProcessSettings.bOverride_LumenFrontLayerTranslucencyReflections = true;
+    PostProcessSettings.LumenFrontLayerTranslucencyReflections = false;
+    PostProcessSettings.bOverride_LumenMaxReflectionBounces = true;
+    PostProcessSettings.LumenMaxReflectionBounces = 1;
+
+    return;
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/SceneCaptureSensor.cpp:767]: The code after this return will never be executed. I suggest you to remove the return or remove the above code for code clarity.

**Review by MarcelPiNacy-CVC — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/carla-simulator/carla/7705)* status: 8 of 11 files reviewed, 7 unresolved discussions (waiting on @Blyron and @xavisolesoft)

___
*[`Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/SceneCaptureSensor.cpp` line 61 at r2](https://reviewable.io/reviews/carla-simulator/carla/7705#gh-1615820365:-Nyzwabm7Z-4GAbIwUyB:b-896fix) ([raw file](https://github.com/carla-simulator/carla/blob/f21d5bedd642da088e9a6d1893266002bfacc99f/Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/SceneCaptureSensor.cpp#L61)):*
<details><summary><i>Previously, Blyron (Blyron) wrote…</i></summary><blockquote>

Remove the commentary
</blockquote></details>

Done.
___
*[`Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/SceneCaptureSensor.cpp` line 808 at r2](https://reviewable.io/reviews/carla-simulator/carla/7705#gh-1615820815:-Nyzwd-gEhYy8p4qGuq5:b-896fix) ([raw file](https://github.com/carla-simulator/carla/blob/f21d5bedd642da088e9a6d1893266002bfacc99f/Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/SceneCaptureSensor.cpp#L808)):*
<details><summary><i>Previously, Blyron (Blyron) wrote…</i></summary><blockquote>

EOF
</blockquote></details>

Done.
___
*[`Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/SceneCaptureSensor.cpp` line 767 at r7](https://reviewable.io/reviews/carla-simulator/carla/7705#gh-1617123207:-NyzwbLH1T3L8lxpp7zR:b-896fix) ([raw file](https://github.com/carla-simulator/carla/blob/a6ad4d9fdd8b9cbbc64bb953692ac5f3475dada0/Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/SceneCaptureSensor.cpp#L767)):*
<details><summary><i>Previously, xavisolesoft (Xavier Solé Nogués) wrote…</i></summary><blockquote>

The code after this return will never be executed. I suggest you to remove the return or remove the above code for code clarity.
</blockquote></details>

Done.
___
*[`Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.h` line 46 at r6](https://reviewable.io/reviews/carla-simulator/carla/7705#-Nyz-49ZDn4oMgVSbQaE:-NyzwdcO7aqcvUd5bmKI:b-896fix) ([raw file](https://github.com/carla-simulator/carla/blob/1d1d13972097a0a1e9c82797dc2bb8f78cd3570b/Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.h#L46)):*
<details><summary><i>Previously, Blyron (Blyron) wrote…</i></summary><blockquote>

Add EOF
</blockquote></details>

Done.
___
*[`Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.cpp` line 13 at r6](https://reviewable.io/reviews/carla-simulator/carla/7705#-Nyz-7nZ8kLcBDUSIqGZ:-NyzweMC9gbzCN0JhQmk:b-896fix) ([raw file](https://github.com/carla-simulator/carla/blob/1d1d13972097a0a1e9c82797dc2bb8f78cd3570b/Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.cpp#L13)):*
<details><summary><i>Previously, Blyron (Blyron) wrote…</i></summary><blockquote>

The indentation must be spaces, not tabs
</blockquote></details>

Done.
___
*[`Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.cpp` line 15 at r6](https://reviewable.io/reviews/carla-simulator/carla/7705#-Nyz-IagAGud6lfUcx6A:-Nyzwf6S6FqpZcLDkUAk:b-896fix) ([raw file](https://github.com/carla-simulator/carla/blob/1d1d13972097a0a1e9c82797dc2bb8f78cd3570b/Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.cpp#L15)):*
<details><summary><i>Previously, Blyron (Blyron) wrote…</i></summary><blockquote>

This is not standard coding. 
The standard coding for UE in component intialization must be like
AClass::AClass(const FObjectIni... ObjInit) :Super(ObjInit)
{
 ComponentVar = CreateDefaultSubbject<UClass>("ComponentVar")
}
</blockquote></details>

Done.

**Review by Blyron — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/carla-simulator/carla/7705)* status: all files reviewed, 8 unresolved discussions (waiting on @MarcelPiNacy-CVC and @xavisolesoft)

___
*[`Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.cpp` line 22 at r8](https://reviewable.io/reviews/carla-simulator/carla/7705#-Nz-8G3_4Sm122T_gPsP:-Nz-8G3_4Sm122T_gPsQ:b8gbwrb) ([raw file](https://github.com/carla-simulator/carla/blob/38f68429070c01af76fc7a3585a3ed06b91467a8/Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Weather/Sky.cpp#L22)):*
> ```cpp
> 	SkyAtmosphereComponent(nullptr)
> {
>   PostProcessComponent = CreateDefaultSubobject<UPostProcessComponent>("ASkyBase::PostProcessComponent");
> ```

Remove ASkyBase from the name for correct visualization in Editor

**Review by xavisolesoft — APPROVED** [review]

**Review by Blyron — COMMENTED** [review]
> Reviewed 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/carla-simulator/carla/7705)* status: all files reviewed, 2 unresolved discussions

**Pull request merged**